marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java

```diff
@@ -86,19 +90,26 @@ public Page<Product> findProducts(String type, String keyword, String language,
     final var typeOption = TypeOption.of(type);
     final var searchPageable = refinePagination(language, pageable);
     Page<Product> result = Page.empty();
+    var searchCriteria = new ProductSearchCriteria();
     switch (typeOption) {
     case ALL:
       if (StringUtils.isBlank(keyword)) {
-        result = productRepository.findAll(searchPageable);
+        result = productRepository.findAllListed(searchPageable);
       } else {
-        result = productRepository.searchByNameOrShortDescriptionRegex(keyword, language, searchPageable);
+        searchCriteria.setKeyword(keyword);
+        searchCriteria.setLanguage(language);
+        result = productRepository.searchListedByCriteria(searchCriteria, searchPageable);
       }
       break;
     case CONNECTORS, UTILITIES, SOLUTIONS:
       if (StringUtils.isBlank(keyword)) {
-        result = productRepository.findByType(typeOption.getCode(), searchPageable);
+        searchCriteria.setType(typeOption);
+        result = productRepository.searchListedByCriteria(searchCriteria, searchPageable);
```

Comment:
please take this line outside and modify if clause to String noneBlank so that codes don't duplicate